File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/util/BackstackHolder.java
package com.zhuinden.simplestackdemoexamplefragments.util;

import com.zhuinden.simplestack.Backstack;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Owner on 2017. 01. 27..
 */
@Singleton
public class BackstackHolder {
    private Backstack backstack;

    @Inject
    public BackstackHolder() {
    }

    public Backstack getBackstack() {
        return this.backstack;
    }

    public void setBackstack(Backstack backstack) {
        this.backstack = backstack;
    }
}

File: examples/legacy-architecture-samples/simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/util/BackstackHolder.kt
package com.zhuinden.simplestackdemoexamplefragments.util

import com.zhuinden.simplestack.Backstack

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Zhuinden on 2018. 08. 20.
 */
@Singleton
class BackstackHolder @Inject constructor() {
    lateinit var backstack: Backstack
}
